### PR TITLE
Feat: Home Page UI & Categories

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added Categories UI
- Categories from API
- Added Video List
- Added get Videos From API

**Files To Check**

- pages/Home
- components/category
- components/sidenav
- components/header
- reducers/video.ts
- sagas/video.ts
- types/video.ts
- actions/video.ts

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,11 +2,13 @@ import { Route, Routes } from "react-router-dom";
...
+import Home from "./Pages/Home";
 
 function App() {
   return (
     <div className="App">
       <Routes>
+        <Route path="/" element={<Home />} />
         <Route path="/login" element={<Login />} />
         <Route path="/register" element={<Register />} />
       </Routes>
```
Comment (shivanigangadharan) [inline src/App.tsx:14]: Suggestion: can create an AppRoutes.js file and import from there as in the future no. of routes will increase so will help in making code more readable here.

File: src/Pages/Home/index.tsx [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+        chart: "mostPopular",
+        maxResults: 10,
+        videoCategoryId: "10",
+      })
+    );
+  }, []);
+
+  return (
+    <HomeContainer>
+      {/* <Carousel items={videos.items} /> */}
```
Comment (shivanigangadharan) [inline src/Pages/Home/index.tsx:32]: Can remove comment if not reqd

File: src/components/Carousel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      className="carousel-container"
+      showThumbs={false}
+      showIndicators={false}
+      showStatus={false}
+    >
+      {items.map((item) => (
+        <>
+          <img
+            className="featured-img"
+            src={item.snippet.thumbnails.high.url}
```
Comment (shivanigangadharan) [inline src/components/Carousel/index.tsx:19]: should give an `alt` attribute here for `img`

File: src/components/VideoList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}) {
+  return (
+    <div className="video-list-container">
+      <div className="video-list-title">{title}</div>
+      <div className="video-list">
+        {items.map((video) => (
+          <Link to={`/video/${video.id}`} className="video">
+            <img
+              src={video.snippet.thumbnails.high.url}
+              className="video-thumbnail"
```
Comment (shivanigangadharan) [inline src/components/VideoList/index.tsx:23]: alt attribute here too

**Review by shivanigangadharan — APPROVED** [review]
Made few small suggestions, other than that code looks well structured. Good job on the UI too. 👍🏼

**Review by ajay-mandaviya — APPROVED** [review]
everything looks good to merge

**Pull request merged**